entitlement/src/main/java/org/killbill/billing/entitlement/dao/BlockingStateSqlDao.java

```diff
@@ -45,9 +46,14 @@ public abstract BlockingStateModelDao getBlockingStateForService(@Bind("blockabl
 
     @SqlQuery
     public abstract List<BlockingStateModelDao> getBlockingState(@Bind("blockableId") UUID blockableId,
+                                                                 @Bind("type") BlockingStateType blockingStateType,
                                                                  @Bind("effectiveDate") Date effectiveDate,
                                                                  @SmartBindBean final InternalTenantContext context);
 
+    @SqlQuery
+    public abstract List<BlockingStateModelDao> getBlockingAllUpToForAccount(@Bind("effectiveDate") Date effectiveDate,
```

Comment:
Thought for further improvements (No action *required* ): Since this api actually fetches everything -- minus the `effectiveDate` restriction -- It could make sense to potentially cache such entries, in such as way that as we call the other already existing apis, we don't make extra calls.